app/client/components/GridView.js

```diff
@@ -902,6 +904,34 @@ GridView.prototype.insertColumn = async function(colId = null, options = {}) {
   return newColInfo;
 };
 
+GridView.prototype.makeHeadersFromRow = async function(selection) {
+  if (this._getCellContextMenuOptions().disableMakeHeadersFromRow){
+    return;
+  }
+  const record = this.tableModel.tableData.getRecord(selection.rowIds[0]);
+  const actions = this.viewSection.viewFields().peek().reduce((acc, field) => {
+    const col = field.column();
+    const colId = col.colId.peek();
+    let formatter = field.formatter();
+    let newColLabel = record[colId];
+    // Manage column that are references
+    if (col.refTable()) {
+      const refTableDisplayCol = this.gristDoc.docModel.columns.getRowModel(col.displayCol());
+      newColLabel =  record[refTableDisplayCol.colId()];
+      formatter = field.visibleColFormatter();
+    }
+    // Manage column that are lists
+    if (isList(newColLabel)) {
+      newColLabel = newColLabel[1];
+    }
+    if (newColLabel) {
```

Comment:
All looks good. Just one last thing I noticed: this condition excludes some value we may want to set, like `0` and `false`.

Our next sync to grist-core will include the following helper in `gristTypes.ts`. If you want to add it to that file now as part of this PR, we can deal with the conflict on our end later.

```javascript
/**
 * Returns true if `value` is blank.
 *
 * Blank values include `null`, (trimmed) empty string, and 0-length lists and
 * reference lists.
 */
export function isBlankValue(value: CellValue) {
  return (
    value === null ||
    (typeof value === 'string' && value.trim().length === 0) ||
    isEmptyList(value) ||
    isEmptyReferenceList(value)
  );
}
```

```suggestion
    if (isBlankValue(newColLabe)l) {
```

Just note that this doesn't handle `undefined`, which can slip in from the `newColLabel = newColLabel[1]` line earlier. I'd maybe change that one to coalesce to `null` like so: `newColLabel = newColLabel[1] ?? null`.

Reply:
I just merged what is it in main and there is no `isBlankValue` helper. I can check explicitly if the value is different of `null` `undefined` or `""`
I think the main use case (maybe 99% of use case) is on text value, I don't know if anyone would like to change the header label to `false` or `0`. If you think it's clearer to do this change I do it :)

Reply:
Sorry, the PR with that change is still in review internally. I don't think it's worth waiting another week for it so your idea to just inline a check is fine (I'd just recommend trimming any strings, as cells can appear blank but contain whitespace).

I agree that use cases would be limited, but I think I'd prefer if we did it anyway if only for the sake of consistency; it's probably not going to be obvious to a user why 0 is ignored but 0.1 is not, etc.